src/V3Param.cpp

```diff
@@ -684,8 +689,25 @@ class ParamProcessor final : public VNDeleter {
                 any_overridesr = true;
             } else {
                 V3Const::constifyParamsEdit(pinp->exprp());
-                AstConst* const exprp = VN_CAST(pinp->exprp(), Const);
-                const AstConst* const origp = VN_CAST(modvarp->valuep(), Const);
+                AstConst* exprp = VN_CAST(pinp->exprp(), Const);
+                AstConst* origp = VN_CAST(modvarp->valuep(), Const);
+                // String constants are parsed as logic arrays and converted to strings in V3Const.
+                // At this moment, some constants may have been already converted.
+                // To correctly compare constants, both should be of the same type,
+                // so they need to be converted.
+                if (exprp && isString(modvarp->subDTypep())) {
+                    pinp->exprp()->replaceWith(new AstConst{exprp->fileline(), AstConst::String{},
+                                                            exprp->num().toString()});
```

Comment:
But it might already be a string.  I'd suggest make a "toString" function and use that here and also right below where you do similar conversion if needed.

Reply:
Done. Unfortunately, the testcase with the default value failed. I will check what is wrong.